tests/Unit/Evolution/Systems/Ccz4/Test_TimeDerivative.cpp

```diff
@@ -0,0 +1,1244 @@
+                        Spectral::Quadrature::GaussLobatto};
+  const auto coord_map =
+      domain::make_coordinate_map<Frame::ElementLogical, FrameType>(Affine3D{
+          Affine{-1., 1., lower_bound[0], upper_bound[0]},
+          Affine{-1., 1., lower_bound[1], upper_bound[1]},
+          Affine{-1., 1., lower_bound[2], upper_bound[2]},
+      });
+  const size_t num_points_3d = num_points_1d * num_points_1d * num_points_1d;
+  const DataVector used_for_size =
+      DataVector(num_points_3d, std::numeric_limits<double>::signaling_NaN());
+  // Setup coordinates
+  const auto x_logical = logical_coordinates(mesh);
+  const auto x = coord_map(x_logical);
+  // Arbitrary time for time-independent solution.
+  const double t = std::numeric_limits<double>::signaling_NaN();
+  // Evaliuate analytic solution
+  const auto kerrschild_vars = solution.variables(
+      x, t, typename gr::Solutions::KerrSchild::tags<DataVector>{});
+
+  // Get ingredients for computing arguments to Ccz4::TimeDerivative
+  const auto& spatial_metric =
+      get<gr::Tags::SpatialMetric<SpatialDim, FrameType, DataVector>>(
+          kerrschild_vars);
+  const auto& d_spatial_metric =
+      get<Tags::deriv<gr::Tags::SpatialMetric<SpatialDim>,
+                      tmpl::size_t<SpatialDim>, FrameType>>(kerrschild_vars);
+  const auto det_and_inverse = determinant_and_inverse(spatial_metric);
+  const auto det_spatial_metric = det_and_inverse.first;
+  const auto d_det_spatial_metric =
+      get<gr::Tags::DerivDetSpatialMetric<SpatialDim, FrameType>>(
+          solution.variables(
+              x, t,
+              tmpl::list<
+                  gr::Tags::DerivDetSpatialMetric<SpatialDim, FrameType>>{}));
+  const auto& dt_spatial_metric =
+      get<Tags::dt<gr::Tags::SpatialMetric<SpatialDim>>>(kerrschild_vars);
+  const auto& inverse_spatial_metric =
+      get<gr::Tags::InverseSpatialMetric<SpatialDim, FrameType, DataVector>>(
+          kerrschild_vars);
+  const auto& lapse = get<gr::Tags::Lapse<DataVector>>(kerrschild_vars);
+  const auto& d_lapse =
+      get<Tags::deriv<gr::Tags::Lapse<DataVector>, tmpl::size_t<SpatialDim>,
+                      FrameType>>(kerrschild_vars);
+  const auto& shift =
+      get<gr::Tags::Shift<SpatialDim, FrameType, DataVector>>(kerrschild_vars);
+  const auto& d_shift =
+      get<Tags::deriv<gr::Tags::Shift<SpatialDim, FrameType, DataVector>,
+                      tmpl::size_t<SpatialDim>, FrameType>>(kerrschild_vars);
+
+  // Params
+  const double c = 1.0;
+  const double cleaning_speed = 1.6;  // e
+  const double f = 0.75;
+  const double kappa_1 = 0.1;
+  const double kappa_2 = 0.3;
+  const double kappa_3 = 0.4;
+  const double mu = 0.7;
+  const double one_over_relaxation_time = 10.0;         // \tau^{-1}
+  const Scalar<DataVector> slicing_condition =          // g(\alpha)
+      get_slicing_condition(use_harmonic_slicing_condition, lapse);
+
+  // Choose free variables \Theta, K_0, b^i, and \eta
```

Comment:
Is theta "free"? I thought it was part of the 4-index constraint, so doesn't it have to be zero? If it doesn't, you would probably want it to be non-zero for testing

Reply:
I may not be using the right terminology, but I thought it was "free" in that you can choose what it is, but correct if if I'm wrong on this. In both test cases, I set it to 0.

Reply:
I looked back at the paper, and yes it seems to be part of the Z4 constraint. In that case, should I not describe it as free?
Let me know if you think I should give it a non-zero value in one or both test cases

Reply:
Well it's not free in the sense of it has to be zero at the beginning to satisfy the Einstein equations. But you could set it to anything you want in a test. Either way, don't worry about it, not worth the effort in my opinion :)